### ISSUE TITLE
Forum Avatar Images Can't Be Blocked

### ISSUE THREAD

**Issue by GABowers** [uBlockOrigin/uBlock-issues#1168]
### Prerequisites

- [x] I verified that this is not a filter issue
    - Filter issues MUST be reported at [filter issue tracker](https://github.com/uBlockOrigin/uAssets/issues)
    
- [x] This is not a support issue or a question
    - Support issues and questions are handled at [/r/uBlockOrigin](https://old.reddit.com/r/uBlockOrigin/)
    
- [x] I performed a cursory search of the issue tracker to avoid opening a duplicate issue
    - Your issue may already be reported.
- I tried to reproduce the issue when...
    - [x] uBlock Origin is the only extension
    - [x] uBlock Origin with default lists/settings
    - [x] using a new, unmodified browser profile
- [x] I am running the latest version of uBlock Origin
- [x] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I report is not a normal behavior

### Description

Forum "avatar" images cannot be blocked

### A specific URL where the issue occurs

forum.unity.com (happens at other forums)

### Steps to Reproduce

1. Go to a forum thread with multiple pages (https://forum.unity.com/threads/why-jobs-just-make-c-tests-and-not-unity-tests.928482/)
2. Use element picker to block avatar image. Example: "||forum.unity.com/data/avatars/m/105/105706.jpg$image"
3. Go to next page in forum thread

### Expected behavior:

The blocked element should disappear.

### Actual behavior:

The element is not blocked, and using the element picker again creates a filter identical to the previous one.

### Your environment

* uBlock Origin version:
1.28.2
* Browser Name and version:
Firefox 78.0.2
* Operating System and version:
Windows 10 Version	10.0.18363 Build 18363

**Comment by GABowers** [comment]
I'm an idiot: this is a filter issue.

**Comment by uBlock-user** [comment]
This happens because the unblocked image was already cached, overiding the cache when refreshing instead of creating a new filter helps.

**Comment by uBlock-user** [comment]
Duplicate of https://github.com/uBlockOrigin/uBlock-issues/issues/1112

**Comment by GABowers** [comment]
Well huh. I tried that before (I tried bypassing it through the uMatrix refresh) and it didn't work, but now it is. Sorry for the duplicate.

**Comment by GABowers** [comment]
Unfortunately the issue is still here.

![filtering_1](https://user-images.githubusercontent.com/15266221/88566698-615aab00-d004-11ea-9031-f2e3164aa115.png)

See the in-focus line 61 on the right window, and the "network filter" address in the picker on the left window. A filter from two days ago isn't blocking the element (if it was blocked, the image wouldn't be visible). Refreshing with Ctrl+F5 does not solve the issue.

As you can see by the tab color I use Firefox containers; the issue persists whether or not I use a container.

**Comment by uBlock-user** [comment]
@GABowers URL ?

**Comment by GABowers** [comment]
https://forum.unity.com/threads/unite-now-optimization-tips-for-maximum-performance-part-1.909794/
Sorry.

**Comment by gwarser** [comment]
What you see in the logger for this network request?

**Comment by gorhill** [comment]
> Unfortunately the issue is still here

It works fine on my side. Always use the logger to disagnose such issue, likely there is something on your side interfering with the block filter.

**Comment by uBlock-user** [comment]
I do see it only I don't use the **refresh** button, but once I click refresh, old cache is bypassed.

**Comment by uBlock-user** [comment]
@GABowers Is _my filters_ activated in Filter lists pane ?

**Comment by GABowers** [comment]
> 
> 
> @GABowers Is _my filters_ activated in Filter lists pane ?

...wow, no. Thanks.

I just created a filter with the full URL (with the part after the file extension) and it worked with that unchecked. With that checked, however, the original filter works.

**Comment by gorhill** [comment]
> and it worked with that unchecked

It can't. If you mean the element picker hid it after you created it, then yes, the element picker will apply the filter immediately regardless of whether your filters are enabled. But it won't apply on reload.

**Comment by uBlock-user** [comment]
Thought so. Your logger screenshot shows the blocking network filter is not getting applied by uBO and that only happens in few cases like having a `$badfilter` of that network filter or when my filters is deactivated by the user.

**Comment by GABowers** [comment]
> 
> 
> > and it worked with that unchecked
> 
> It can't. If you mean the element picker hid it after you created it, then yes, the element picker will apply the filter immediately regardless of whether your filters are enabled. But it won't apply on reload.

What do you mean by "reload"? When I uncheck "my filters", use the element picker to pick an image and filter it, then refresh the page with F5 _or_ Ctrl+F5, it's still not visible.

Either way, PEBCAK. Appreciate the help.
